### PR TITLE
Add initial tests for the support_panel function

### PR DESCRIPTION
**Pull request by cjrace**
These are the first unit tests we've added so hopefully lay an example for how to add them.

Added basic validation (can be improved in future!) to the team_email argument in the support_panel() function.

I used usethis::use_test("support-panel") to set up the scripts for tests and then added a few simple test case emails in there for ones that should pass, and ones that should trigger the stop message.

----

Edit:

This adds R CMD Check as a way to run the tests and test the package more widely using github actions. This also adds a code coverage action using https://app.codecov.io/. That will likely only start running once this branch is merged into main

### CODE REVIEW

File: R/standard_panels.R [hunk 1]
```diff
@@ -26,6 +26,21 @@ support_panel <- function(
     publication_stub = "",
     alt_href = NULL,
     form_url = NULL) {
+  # Check that the team_email is a valid email
+
+  is_valid_email <- function(email) {
+    grepl(
+      "\\<[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,}\\>",
```
Comment (rmbielby) [inline R/standard_panels.R:33]: Can we be more stringent than this? Should we mandate that it should specifically be an @education.gov.uk address?

File: R/standard_panels.R [hunk 2]
```diff
@@ -26,6 +26,21 @@ support_panel <- function(
...
+      as.character(email),
+      ignore.case = TRUE
+    )
+  }
+
+  if (is_valid_email(team_email) == FALSE) {
+    stop("You have entered an invalid email in the team_email argument.")
```
Comment (rmbielby) [inline R/standard_panels.R:40]: If we do go more stringent, would be good to define in this error messaging what we're expecting see.

**Comment by rmbielby** [comment]
Any reason for not setting up a workflow to run any tests on PRs?

**Comment by cjrace** [comment]
> Any reason for not setting up a workflow to run any tests on PRs?

The workflow I think we should set up is the general R CMD check, that runs the tests amongst other things rather than just having one for our automated tests. 

Happy to add that here or in a separate branch with other workflows?

File: R/standard_panels.R (hunk 1 above)
Comment (cjrace) [reply R/standard_panels.R:33]: Good point, I'll add this in!

**Comment by cjrace** [comment]
Closing in favour of a new PR

**Pull request closed without merging**